resources/codegen_inputs/meta.rs

```diff
@@ -0,0 +1,38 @@
+#![parse_module(read_fonts::tables::meta)]
+
+/// [`meta`](https://docs.microsoft.com/en-us/typography/opentype/spec/meta)
+#[tag = "meta"]
+table Meta {
+    /// Version number of the metadata table — set to 1.
+    #[compile(1)]
+    version: u32,
+    /// Flags — currently unused; set to 0.
+    #[compile(0)]
+    flags: u32,
+    /// Not used; set to 0.
+    #[skip_getter]
+    #[compile(0)]
+    reserved: u32,
+    /// The number of data maps in the table.
+    #[compile(array_len($data_maps))]
```

Comment:
There is one part here that I wanted to gather opinion on for the initial implementation :mag_right:

[`array_len` mandates that its calculated value fits within `u16`](https://github.com/googlefonts/fontations/blob/a41651ce0ee061bbda491f011894fd183d053ff0/write-fonts/src/lib.rs#L185-L188); it is unlikely that any production fonts are compiling `meta` tables with more than ~60,000 records, but is it worth fontations using a split or generic `array_len` function to fully support these wider types? (from memory, this case appears in a few other codegen inputs too)

Reply:
good catch! I had been thinking that array_len returned a usize, so if this is returning a `u16` it's definitely not right, let me see if there's an easy fix.

Reply:
I've opened #1251 to fix the issue with array_len, which as you note predates this PR, so I'll go ahead and merge this. :)